test/test_vocab.py

```diff
@@ -1,33 +1,126 @@
-import numpy as np
-import unittest
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from collections import Counter
+import os
+
 
+import numpy as np
+from numpy.testing import assert_allclose
+import torch
 from torchtext import vocab
-from collections import Counter
 
+from .common.test_markers import slow
+from .common.torchtext_test_case import TorchtextTestCase
+
+
+class TestVocab(TorchtextTestCase):
+    def test_vocab_basic(self):
+        c = Counter({'hello': 4, 'world': 3, 'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T': 5, 'freq_too_low': 2})
+        v = vocab.Vocab(c, min_freq=3, specials=['<pad>', '<bos>'])
+
+        self.assertEqual(v.itos, ['<unk>', '<pad>', '<bos>',
+                                  'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T', 'hello', 'world'])
+
+    def test_vocab_set_vectors(self):
+        c = Counter({'hello': 4, 'world': 3, 'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T': 5,
+                     'ｔｅｓｔ': 4, 'freq_too_low': 2})
+        v = vocab.Vocab(c, min_freq=3, specials=['<pad>', '<bos>'])
+        stoi = {"hello": 0, "world": 1, "ｔｅｓｔ": 2}
+        vectors = torch.FloatTensor([[0.1, 0.2], [0.3, 0.4], [0.5, 0.6]])
+        dim = 2
+        v.set_vectors(stoi, vectors, dim)
+        expected_vectors = np.array([[0.0, 0.0], [0.0, 0.0], [0.0, 0.0],
+                                     [0.0, 0.0], [0.1, 0.2], [0.5, 0.6],
+                                     [0.3, 0.4]])
+        assert_allclose(v.vectors.numpy(), expected_vectors)
+
+    def test_vocab_download_fasttext_vectors(self):
+        c = Counter({'hello': 4, 'world': 3, 'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T': 5, 'freq_too_low': 2})
+        # Build a vocab and get vectors twice to test caching.
+        for i in range(2):
+            v = vocab.Vocab(c, min_freq=3, specials=['<pad>', '<bos>'],
+                            vectors='fasttext.simple.300d')
+
+            self.assertEqual(v.itos, ['<unk>', '<pad>', '<bos>',
+                                      'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T', 'hello', 'world'])
+            vectors = v.vectors.numpy()
+
+            # The first 5 entries in each vector.
+            expected_fasttext_simple_en = {
+                'hello': [0.39567, 0.21454, -0.035389, -0.24299, -0.095645],
+                'world': [0.10444, -0.10858, 0.27212, 0.13299, -0.33165],
+            }
+
+            for word in expected_fasttext_simple_en:
+                assert_allclose(vectors[v.stoi[word], :5],
+                                expected_fasttext_simple_en[word])
+
+            assert_allclose(vectors[v.stoi['<unk>']], np.zeros(300))
+        # Delete the vectors after we're done to save disk space on CI
+        if os.environ.get("TRAVIS") == "true":
+            os.remove(os.path.join(self.project_root, ".vector_cache",
```

Comment:
Can you add a check to remove files only if they exist? As is, other tests will fail if they also try to remove the same file.

Reply:
I can also just add this in https://github.com/pytorch/text/pull/115 since that is where it is tripping me up right now.

Reply:
@bmccann out right now, so it might be a few hours. Since you have write access to the pro, you should also have write permissions on this PR --- feel free to go ahead and make the change if I haven't gotten to it.

Reply:
whoops, just saw that you added it in #115, that's fine with me